### DISCUSSION TITLE
General Discussion on hot reloading when developing custom nodes

### DISCUSSION THREAD

**Discussion by regiellis** [Comfy-Org/ComfyUI#5290] [General]
Hi All,

Just started writing custom nodes for Comfy and wanted to get a general feeling on how others are handling the reload
on change situation, and if there is a official solution coming? At first I was a little frustrated with process and decided to
write my own [hack watcher due to my custom bootstrap](https://gist.github.com/regiellis/7406d62ac47b6e9926b3231d34d0a553), which works well enough for me and is straight forward. There is also [ComfyUI-HotReloadHack](https://github.com/logtd/ComfyUI-HotReloadHack)...but again I am wondering if there is a) a better way or b) as mention a official solution coming?

Be nice to hear what others are doing: running and developing comfy on a custom AI Ubuntu box I built...so I dont even know
how bad it is on windows.

**Comment by ltdrdata** [comment]
Hot reloading has been discussed for a very long time. While it can be used for debugging simple situations, it cannot be applied for end users due to many complex situations (especially issues with imported dependencies).

FYI, https://github.com/comfyanonymous/ComfyUI/pull/749

**Reply by regiellis** [reply]
Understood, I have modified my scripts to work with the CLI in my environment and it has been working well for me right now. Thanks for the issue link.
